phased-updates: first pass of technical updates

Hyask opened the pull request:
That documentation is written like a live document, more or less playing the role of a bug tracker with open questions and future work. This doesn't fit a documentation page, so I've removed most of those entries, and corrected what I could so that at least the technical information in that page is correct. This still needs an overhaul to better fit modern documentation standards.

cc: @s-makin

rkratky reviewed (commented):
Thanks, @Hyask. A few suggestions and a question.

s-makin commented on docs/staging/phased-updates.md:
```diff
@@ -89,101 +81,10 @@ package and version at a particular threshold. It does vary based on package
 
-* **Launchpad entry**: [Phased updates of software packages](https://blueprints.launchpad.net/ubuntu/+spec/foundations-r-phased-updates)
+* **Launchpad spec entry**: [Phased updates of software packages](https://blueprints.launchpad.net/ubuntu/+spec/foundations-r-phased-updates)
 * **Originally created**: 2012-05-14
 * **Last updated**: 2023-06-09
```
I don't think it's necessary, it's another artifact left behind in the migration

Hyask commented:
Thanks both for the review. I've addressed your comments and suggestion, added a link to the `phased-updater` script, and dropped a few blocks.
I still think this would need some proper structure from a technical author, as right now, this still feel like half a spec, half a bug tracker, and another half of end-user documentation. That many halves can't be good for sure :grinning:
